### ISSUE TITLE
google_compute_address removal and cost calculation

### ISSUE THREAD

**Issue by gtirloni** [infracost/infracost#1619]
Whenever we remove a public compute address from a VM in GCP, infracost shows this:

```
- google_compute_address.my-instance
  -$11.68

    - IP address (if used by standard VM)
      -$2.92

    - IP address (if used by preemptible VM)
      -$1.46

    - IP address (if unused)
      -$7.30
```

Then it sums all these 3 costs and says our total cost will decrease by $11.68, which does not make much sense.

The only cost decrease we are expecting is the $2.92 for a in-use IP address and nothing else.

How can we improve infracost's understanding in this case?

**Comment by vdmgolub** [comment]
@gtirloni Hello! Thank you for creating the issue. It really doesn't look right.

As an immediate fix you can try specifying a usage param `address_type` for the IP address resources. We've added it in the latest [v0.9.23](https://github.com/infracost/infracost/issues/1572) version. Setting it in your infracost-usage.yml file should show only one cost component per resource:
```yml
  google_compute_address.my_address:
    address_type: standard_vm # IP address being used, can be: standard_vm, preemptible_vm, unused.
```

```
└─ IP address (standard VM)                                                    730  hours                  $2.92 
```

Will this help your case?

**Comment by tim775** [comment]
@gtirloni Thanks for pointing this out, I agree the current implementation seems confusing to me too.  There was recent work done to this in #1573 but I'm wondering if we can detect the connection status from the terraform.  Is your use case basically the same as the [Instance With IP](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_address#example-usage---instance-with-ip) example in the Terraform docs?  If not maybe you could put together a sanitized example tf?

**Comment by gtirloni** [comment]
@tim775 yep, that's exactly the case. I'll update this issue with a proper example later.

**Comment by vdmgolub** [comment]
@gtirloni Hey! We've just merged a fix for the Compute Address resource: from now on we try to detect how the IP address is used (VM type, is it used or not) and we'll show only one cost component, so there should not be multiple costs. We also removed the `address_type` usage param as redundant because we detect the usage now.

This should go live with the next release. Thank you for reporting the issue! :)
